include/cuco/detail/dynamic_map.inl

```diff
@@ -80,25 +121,24 @@ void dynamic_map<Key, Value, Scope, Allocator>::reserve(std::size_t n)
 
 template <typename Key, typename Value, cuda::thread_scope Scope, typename Allocator>
 template <typename InputIt, typename Hash, typename KeyEqual>
-void dynamic_map<Key, Value, Scope, Allocator>::insert(InputIt first,
-                                                       InputIt last,
-                                                       Hash hash,
-                                                       KeyEqual key_equal)
+void dynamic_map<Key, Value, Scope, Allocator>::insert(
+  InputIt first, InputIt last, Hash hash, KeyEqual key_equal, cudaStream_t stream)
 {
   std::size_t num_to_insert = std::distance(first, last);
-  reserve(size_ + num_to_insert);
+
+  reserve(size_ + num_to_insert, stream);
 
   uint32_t submap_idx = 0;
   while (num_to_insert > 0) {
     std::size_t capacity_remaining =
       max_load_factor_ * submaps_[submap_idx]->get_capacity() - submaps_[submap_idx]->get_size();
     // If we are tying to insert some of the remaining keys into this submap, we can insert
     // only if we meet the minimum insert size.
+
     if (capacity_remaining >= min_insert_size_) {
-      *num_successes_ = 0;
-      int device_id;
-      CUCO_CUDA_TRY(cudaGetDevice(&device_id));
-      CUCO_CUDA_TRY(cudaMemPrefetchAsync(num_successes_, sizeof(atomic_ctr_type), device_id));
+      // TODO: memset an atomic variable is unsafe
+      static_assert(sizeof(std::size_t) == sizeof(atomic_ctr_type));
+      CUCO_CUDA_TRY(cudaMemset(submap_num_successes_[submap_idx], 0, sizeof(atomic_ctr_type)));
```

Comment:
```suggestion
      CUCO_CUDA_TRY(cudaMemsetAsync(submap_num_successes_[submap_idx], 0, sizeof(atomic_ctr_type), stream));
```
Although I don't know if this plays well with the `CUCO_CUDA_TRY` macro.

Reply:
Hm, why wouldn't it play well with `CUCO_CUDA_TRY`? I think the only way the call can return an error is if the memory address is invalid, which is independent of whether or not the memory has actually been written yet, right?

Reply:
You're right. The async memcpy should return a value immediately which is then consumed by the error macro.